src/cargo_communication/request_actor.rs

```diff
@@ -348,3 +348,837 @@ pub trait CargoHandleTrait<T> {
 
     fn join(self) -> io::Result<ExitStatus>;
 }
+
+#[cfg(test)]
+pub mod compile_request_tests {
+    use std::collections::HashSet;
+    use std::fmt::Debug;
+    use std::hash::Hash;
+    use std::os::unix::prelude::ExitStatusExt;
+    use std::time::Duration;
+
+    use crate::bsp_types::notifications::{
+        CompileReportData, CompileTaskData, Diagnostic as LSPDiagnostic,
+        Notification as NotificationTrait, PublishDiagnostics, PublishDiagnosticsParams,
+        TaskDataWithKind, TaskFinish, TaskFinishParams, TaskId, TaskProgress, TaskProgressParams,
+        TaskStart, TaskStartParams,
+    };
+    use crate::bsp_types::requests::{Compile, CompileParams, CompileResult};
+    use crate::bsp_types::{BuildTargetIdentifier, TextDocumentIdentifier};
+    use crate::cargo_communication::request_actor::CargoMessage::CargoStdout;
+    use crate::communication::{ErrorCode, Message, Notification, Response};
+    use cargo_metadata::diagnostic::{
+        DiagnosticBuilder, DiagnosticCodeBuilder, DiagnosticSpanBuilder, DiagnosticSpanLineBuilder,
+    };
+    use cargo_metadata::Message::{
+        BuildFinished, BuildScriptExecuted, CompilerArtifact, CompilerMessage,
+    };
+    use cargo_metadata::{
+        ArtifactBuilder, ArtifactProfileBuilder, BuildFinishedBuilder, BuildScriptBuilder,
+        CompilerMessageBuilder, Edition, PackageId, TargetBuilder,
+    };
+    use crossbeam_channel::{unbounded, Sender};
+    use lsp_types::{
+        DiagnosticRelatedInformation, DiagnosticSeverity, Location, NumberOrString, Position, Range,
+    };
+    use serde_json::to_string;
+    use url::Url;
+
+    use super::*;
+
+    fn init_test(
+        mut mock_cargo_handle: MockCargoHandleTrait<CargoMessage>,
+    ) -> (Receiver<RPCMessage>, Sender<CargoMessage>, Sender<Event>) {
+        let (sender_to_main, receiver_to_main) = unbounded::<RPCMessage>();
+        let (sender_from_cargo, receiver_from_cargo) = unbounded::<CargoMessage>();
```

Comment:
nie rozumiem co znaczy "sender from" i "receiver to"

Reply:
Znaczenie miało być takie, że sender_to_main to sender którego używamy do wysłania wiadomości, a receiver_to_main to receiver którego używamy w mainie. Faktycznie te nazwy nie są najlepsze więc podmienie je na "receiver_from_actor" i "sender_to_actor".